viewer/packages/rendering/src/rendering/materials.ts

```diff
@@ -25,18 +23,36 @@ export interface Materials {
   instancedMesh: THREE.RawShaderMaterial;
   triangleMesh: THREE.RawShaderMaterial;
   simple: THREE.RawShaderMaterial;
+  texturedMaterials: { [key: string]: THREE.RawShaderMaterial };
+}
+
+export function forEachMaterial(materials: Materials, callback: (material: THREE.RawShaderMaterial) => void): void {
+  callback(materials.box);
+  callback(materials.circle);
+  callback(materials.generalRing);
+  callback(materials.nut);
+  callback(materials.quad);
+  callback(materials.cone);
+  callback(materials.eccentricCone);
+  callback(materials.sphericalSegment);
+  callback(materials.torusSegment);
+  callback(materials.generalCylinder);
+  callback(materials.trapezium);
+  callback(materials.ellipsoidSegment);
+  callback(materials.instancedMesh);
+  callback(materials.triangleMesh);
+  callback(materials.simple);
```

Comment:
Because one of the members of the `materials` object is now a map of textured materials, we don't get a consistent type on the values of the object. We did something similar before (although I don't understand totally why), so I thought this was acceptible

Reply:
Can't we then check for that property specifically and thus convert type used by callback?

Reply:
Alright - I changed it now